File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        StringTokenizer st;

        String[][] array = new String[N][2];

        for(int i = 0; i < N; i++){
            st = new StringTokenizer(br.readLine());
            array[i][0] = st.nextToken();
            array[i][1] = st.nextToken();
        }

        br.close();
        
        Arrays.sort(array, Comparator.comparingInt(a -> Integer.parseInt(a[0])));

        for(int i = 0; i < N; i++){
            System.out.println(array[i][0] + " " + array[i][1]);
        }
    }
}

File: 백준/Silver/10814. 나이순 정렬/나이순 정렬.kt
import java.util.*
import java.io.*

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {

    val N = readLine().toInt()
    val arr = arrayListOf<Pair<Int, String>>()
    for(i in 0 until N) {

        val st = StringTokenizer(readLine())
        val age = st.nextToken().toInt()
        val name = st.nextToken()
        arr.add(Pair(age, name))
    }
    close()

    // 나이 오름차순 정렬
    arr.sortBy { it.first }

    val sb = StringBuilder()
    for(data in arr) {
        sb.append("${data.first} ${data.second}\n")
    }
    println(sb)
}